crates/cast/bin/cmd/run.rs

```diff
@@ -10,14 +10,24 @@ use foundry_cli::{
     update_progress,
     utils::{handle_traces, TraceResult},
 };
-use foundry_common::{is_known_system_sender, SYSTEM_TRANSACTION_TYPE};
+use foundry_common::{is_known_system_sender, SYSTEM_TRANSACTION_TYPE,
+    compile::ProjectCompiler, fs,
+};
 use foundry_compilers::EvmVersion;
 use foundry_config::{find_project_root_path, Config};
 use foundry_evm::{
     executors::{EvmError, TracingExecutor},
     opts::EvmOpts,
     utils::configure_tx_env,
 };
+use std::collections::BTreeMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+struct CachedSignatures {
+    events: BTreeMap<String, String>,
+    functions: BTreeMap<String, String>,
+}
```

Comment:
we already have this type here

https://github.com/foundry-rs/foundry/blob/edb3a4b125510c1b24e40b69218aada87b73489d/crates/evm/traces/src/identifier/signatures.rs#L14-L18

we should prevent duplicating this and instead find a way to reuse the existing type

we can make the type pub instead and also make this a helper function:

https://github.com/foundry-rs/foundry/blob/eef87de35375311f134fb4261c18455ced8022a5/crates/evm/traces/src/identifier/signatures.rs#L44-L56

Reply:
I defined it again because I find it does not exported. For maintenance reasons, I think you're right. At first I was just using it for myself so I didn't think about it that much, I'll spend some time refactoring the code next.